test.lua

```diff
@@ -584,6 +584,40 @@ for k, v in pairs(quoting_tests) do
     end
 end
 
+---- docstring tests ----
+
+print("Running tests for docstrings...")
+local docstring_tests = {
+  ['(doc (fn [] :A 1))'] = {'(_0_)\\n  A', 'anonymous fn docstring mistmatch'},
+  ['(doc (λ [a] :B :ret-str))'] = {'(_0_ a)\\n  B', 'anonymous lambda docstring mistmatch'},
```

Comment:
I'm not sure how I feel about `_0_` showing up here... On the other hand, leaving it out seems weird, and I'm not sure what it would make sense to replace it with... maybe the `tostring` of the function?

`(#<function: 0x5623043dd020> a)\n\n   B`

Reply:
I had the same thought, and figured a subsequent PR could clean that up a bit but in the meantime it was better than not having it at all.

My initial thought was replacing `_n_` with something like `<anonymous>` so it would look like `(<anonymous> a)\n  B` but I can see the benefit of having the address in there.

That said, I feel like if we're going to explicitly handle that case we should make it clear to the end-user that it's showing up that way because there wasn't a function name. Thoughts?

Reply:
Yeah, explicitly indicating it's anonymous is good. What about `#<anonymous>` since the `#<>` notation is already used for "unreadable" things? But we could leave it as-is for now and come back to it later; that's fine.

Reply:
`#<anonymous>` sounds good to me! This will be easy enough to throw into the `doc` function when I update the branch to store `fnName` and `argsList` as separate metadata fields, since the work of assembling the pretty output will be moved to the `doc` function - I can just do something like
```lua
fennel.doc = function doc(fn)
  local docstring = metadata:get(fn, 'docstring') or '<undocumented>'
  local fnName = metadata:get(fn, 'fnName') or '#<anonymous>'
  local argsList = metadata:get(fn, 'argsList') or {}
  local invokeStr = string.format('(%s%s%s)', fnName, #argsList > 0 and ' ' or '', table.concat(argsList, ' '))
  print(string.format('%s\n  %s', invokeStr, docstring))
end
```

Reply:
I tried implementing this for the update I just pushed, but it turned out to be a bit extra confusing due to the realization that functions set to local/global vars or table properties will be treated as anonymous:

E.g.
```clojure
(global foo (fn [] :stuff) (doc foo) ; -> (#<anonymous)\n  <nil>
(local tbl {:foo (fn [] "do things!" :things)})
(doc tbl.foo) ; -> #(anonymous)\n  do things!
```

Detecting those kinds of assignments seems like a nontrivial addition so I'm going to leave it off for now.